src/rust/src/backend/ec.rs

```diff
@@ -34,8 +34,8 @@ fn curve_from_py_curve(
     if !py_curve.is_instance(&types::ELLIPTIC_CURVE.get(py)?)? {
         if allow_curve_class {
             let warning_cls = types::DEPRECATED_IN_42.get(py)?;
-            let warning_msg = "Curve argument must be an instance of an EllipticCurve class. Did you pass a class by mistake? This will be an exception in a future version of cryptography.";
-            pyo3::PyErr::warn_bound(py, &warning_cls, warning_msg, 1)?;
+            let message = "Curve argument must be an instance of an EllipticCurve class. Did you pass a class by mistake? This will be an exception in a future version of cryptography.\0".as_bytes();
+            pyo3::PyErr::warn(py, &warning_cls, _str_ref_to_cstr_ref(message), 1)?;
```

Comment:
maybe instead of the helper I could make a macro that does `concat!(message, "\0").as_bytes()` and then converts that to a &CStr. I see there's only one macro inside cryptography right now - is there a good place to put it besides the crate root?

Reply:
Yeah, I think this is so simple there's limited value, it'd be better to just have the contents inlined.

Reply:
macro in `common.rs` sounds fine